### ISSUE TITLE
reads with polyA tails

### ISSUE THREAD

**Issue by cheny19** [alexdobin/STAR#408]
Dear Alex,

I'm new to STAR and I'm trying to use it to align RNA-seq PolyA+ reads to reference genome. Some of the reads contain a polyA tail and I noticed that this potentially leads to unwanted output. 

For example, I have a read pair:
```
>SN7001282:314:h15b0adxx:1:1101:14617:3004 1:N:0:1
TTTTTTTTTTGACGTTTGATGTTTTATATCATATCAATAACAACGTACCGTACAAAACCTGCCTCCCAGCTCGTG

>SN7001282:314:h15b0adxx:1:1101:14617:3004 2:N:0:1
GCAAAACCCTCTTTCCTCTCCTCTTTTGGGACAAGAGCCCTGGTTTTCTACGCTGCCCTTGGCCACCACACTGCC
```

When I use default parameters to align the read pair to Ensembl GRCh37.75, I get an unique alignment:
```
SN7001282:314:h15b0adxx:1:1101:14617:3004	163	5	122562075	255	75M	=	122562155	155	GCAAAACCCTCTTTCCTCTCCTCTTTTGGGACAAGAGCCCTGGTTTTCTACGCTGCCCTTGGCCACCACACTGCC	@@@DDEDDHHFBFGGIIBGEHEFHIIBGHGFAFBFAGAGH;DBDGCD<4?FGHCFBEGICEFEEE???B?AACCA	NH:i:1	HI:i:1	AS:i:144	nM:i:2
SN7001282:314:h15b0adxx:1:1101:14617:3004	83	5	122562155	255	75M	=	122562075	-155	CACGAGCTGGGAGGCAGGTTTTGTACGGTACGTTGTTATTGATATGATATAAAACATCAAACGTCAAAAAAAAAA	8-B>>BA?<AC@BA=;?BAA>5=;5>A=6?B=@=)CFCC==8)FACHECGHFD?HDBAGBDF>FCDFDDDDD@@@	NH:i:1	HI:i:1	AS:i:144	nM:i:2
```

But when I BLAT read1 and view it in UCSC genome browser, it can actually be aligned to multiple locations. It can be aligned to an annotated gene on chr19 and the polyT seems to be the polyA tail, while on chr5, it can only be aligned to an intergenic region.

I also tried removing the polyTs manually and aligned them again with default parameters, and STAR now can locate them correctly:
```
SN7001282:314:h15b0adxx:1:1101:14617:3004	163	19	18281198	255	75M	=	18281278	145	GCAAAACCCTCTTTCCTCTCCTCTTTTGGGACAAGAGCCCTGGTTTTCTACGCTGCCCTTGGCCACCACACTGCC	@@@DDEDDHHFBFGGIIBGEHEFHIIBGHGFAFBFAGAGH;DBDGCD<4?FGHCFBEGICEFEEE???B?AACCA	NH:i:1	HI:i:1	AS:i:138	nM:i:0
SN7001282:314:h15b0adxx:1:1101:14617:3004	83	19	18281278	255	65M	=	18281198	-145	CACGAGCTGGGAGGCAGGTTTTGTACGGTACGTTGTTATTGATATGATATAAAACATCAAACGTC	8-B>>BA?<AC@BA=;?BAA>5=;5>A=6?B=@=)CFCC==8)FACHECGHFD?HDBAGBDF>FC	NH:i:1	HI:i:1	AS:i:138	nM:i:0
```

My question is, it looks like a multi-alignment to me, but STAR only outputs an unique alignment. Is there a way to get both alignments? For example, for the trimmed read pair, the second alignment should be on chr5 with one mismatch, but why STAR doesn't output that?

Thanks,
Chen

**Comment by alexdobin** [comment]
Hi Chen,

STAR limits the multimappers to those within --outFilterMultimapScoreRange of the best alignment, = 1 by default. An alignment with one mismatch will have a score lower by 2 points than the alignment without mismatches (best), so STAR will not output it. You can try to increase the --outFilterMultimapScoreRange to 2, and you will likely get output of the 2nd alignment. However, this is not guaranteed, as finding alignments with mismatches is harder than without.

Cheers
Alex

**Comment by cheny19** [comment]
Thanks for your prompt response! I tried modifying `--outFilterMultimapScoreRange` and it worked! The output looks like this:

```
@CO	user command line: STAR --genomeDir STAR_db_hg19/ --readFilesIn ../raw_reads/test_R1.fastq ../raw_reads/test_R2.fastq --outFileNamePrefix test --outFilterMultimapScoreRange 10
SN7001282:314:h15b0adxx:1:1101:14617:3004	163	5	122562075	3	75M	=	122562155	155	GCAAAACCCTCTTTCCTCTCCTCTTTTGGGACAAGAGCCCTGGTTTTCTACGCTGCCCTTGGCCACCACACTGCC	@@@DDEDDHHFBFGGIIBGEHEFHIIBGHGFAFBFAGAGH;DBDGCD<4?FGHCFBEGICEFEEE???B?AACCA	NH:i:2	HI:i:1	AS:i:144	nM:i:2
SN7001282:314:h15b0adxx:1:1101:14617:3004	83	5	122562155	3	75M	=	122562075	-155	CACGAGCTGGGAGGCAGGTTTTGTACGGTACGTTGTTATTGATATGATATAAAACATCAAACGTCAAAAAAAAAA	8-B>>BA?<AC@BA=;?BAA>5=;5>A=6?B=@=)CFCC==8)FACHECGHFD?HDBAGBDF>FCDFDDDDD@@@	NH:i:2	HI:i:1	AS:i:144	nM:i:2
SN7001282:314:h15b0adxx:1:1101:14617:3004	419	19	18281198	3	75M	=	18281278	145	GCAAAACCCTCTTTCCTCTCCTCTTTTGGGACAAGAGCCCTGGTTTTCTACGCTGCCCTTGGCCACCACACTGCC	@@@DDEDDHHFBFGGIIBGEHEFHIIBGHGFAFBFAGAGH;DBDGCD<4?FGHCFBEGICEFEEE???B?AACCA	NH:i:2	HI:i:2	AS:i:138	nM:i:0
SN7001282:314:h15b0adxx:1:1101:14617:3004	339	19	18281278	3	65M10S	=	18281198	-145	CACGAGCTGGGAGGCAGGTTTTGTACGGTACGTTGTTATTGATATGATATAAAACATCAAACGTCAAAAAAAAAA	8-B>>BA?<AC@BA=;?BAA>5=;5>A=6?B=@=)CFCC==8)FACHECGHFD?HDBAGBDF>FCDFDDDDD@@@	NH:i:2	HI:i:2	AS:i:138	nM:i:0
```

Now my question is, what's the penalty score for softclipping, or how softclipping affects the final score of an alignment? In my case, it looks like an alignment with 2 mismatches has a much higher score than the alignment with 0 mismatch but softclipped. Since the length of polyA tail in each read is different, the length of softclipped region is also different. How can I set `--outFilterMultimapScoreRange` to ensure these alignments are also ouput?

**Comment by alexdobin** [comment]
Hi Chen,

the actual formula for the alignment score is +1 for matches and -1 for mismatches.
So one mismatch compared to no mismatches is effectively penalized with -2.
Similarly, the soft-clipping is penalized with -1 per soft-clipped bases compared to a matched base.

At the moment there is no way to avoid the penalty for soft-clipped tails. I think the best way to deal with this is to trim A-tail before mapping - this should result in all the alignments reported even with small --outFilterMultimapScoreRange .

Cheers
Alex

**Comment by sjackman** [comment]
Hi, Alex. Is it still recommended to trim polyA tails before alignment? Is there a STAR FAQ?

**Comment by alexdobin** [comment]
Hi Shaun,

if you expect a large number of reads with long A-tails, trimming of the tails before mapping would be beneficial. You can use the built-in adapter trimmer for that, e.g.:
--clip3pAdapterSeq AAAAAAAA --clip3pAdapterMMp 0.1

No STAR FAQ yet, unfortunately...

Cheers
Alex

**Comment by MiriamAng** [comment]
Hi Alex,
I have the need to remove polyA tails from raw reads coming from QuantSeq 3' data that I am processing using the nf-core/rnaseq pipeline. I came across the Slack nf-core/rnaseq channel where it was mentioned the possibility to perform polyA removal using the STAR parameter --clip3pAdapterSeq. I added this parameter to STAR, however it is not clear to me how the length of the As specified influences the trimming. Leaving aside the clip3pAdapterMMp parameter, if I set --clip3pAdapterSeq AAAAAAAA, does it mean that it will remove only a number of As equal to the one specified? If the actual number in the raw read is lower or higher than the number specified, what is the default behavior? I don't know a priori how many As will be in my reads, the number is highly variable, therefore I was wondering whether this parameter can be used also in my case. 
Thank you for your time!
Miriam
